### ISSUE TITLE
Using cosine distance or similarity?

### ISSUE THREAD

**Issue by AmirAlavi** [valerystrizh/pytorch-histogram-loss#8]
Thanks for the implementation. It looks like you are computing a cosine similarity, not the cosine distance (distance = -1* similarity):

https://github.com/valerystrizh/pytorch-histogram-loss/blob/dca1be0123446948ed61dcc8bdc371a14daf9180/losses.py#L35

The above matrix multiplication will give you cosine similarity. I thought that these are supposed to be distances according to the paper? Do you negate them somewhere later?

**Comment by AmirAlavi** [comment]
Ah nevermind, in the paper, distance is only used for visualization in Figure 5, but the actual histogram loss for training using similarities. My mistake. Thanks again!
